### DISCUSSION TITLE
Anyone use this with this with wifi pump model?

### DISCUSSION THREAD

**Discussion by wyliecoyoteuk** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#138] [Ideas]
Just thinking, I have a wifi enabled pump, but the LayZspa app is useless, the Bestway one doesn't support scheduling.
I have fitted this controller to my spare pump, it works fine, so I am thinking of swapping it over and modding the Wifi enabled one to match.
So does anyone know if the wifi pumps have a different interface?

**Comment by visualapproach** [comment]
Hi, does it have the same cable between display and pump? I have no idea what's in that pump, but it would be interesting to know. Let us know if it works. Have a nice day

**Reply by wyliecoyoteuk** [reply]
Looks like it, it has an additional unit next to the control panel, which contains an ESP8266.
Model number is 54184
Some photos:
![IMG_20211109_172224231](https://user-images.githubusercontent.com/79256941/140976884-36b23d06-2d7d-40f9-b7c0-44509d03d017.jpg)
![IMG_20211109_172236337](https://user-images.githubusercontent.com/79256941/140976889-a1e1de8c-93d5-44a3-847c-2342a6b7ffc4.jpg)
![IMG_20211109_172401770](https://user-images.githubusercontent.com/79256941/140976895-bced0921-6d88-4bf5-a9d5-3e14cc4113b6.jpg)
![IMG_20211109_172425043](https://user-images.githubusercontent.com/79256941/140976900-dece4069-02aa-4e69-8ee0-d9786a89b927.jpg)
![IMG_20211109_172819211_HDR](https://user-images.githubusercontent.com/79256941/140976905-bfe20c22-8083-492d-b8f4-02fa3990c0c2.jpg)
![IMG_20211109_172828328_HDR](https://user-images.githubusercontent.com/79256941/140976907-533ef4f9-2b47-4568-9bf2-6d0aba1fb955.jpg)

**Reply by visualapproach** [reply]
Cute little dongle. WROOM is an ESP32 i think. It looks like an add-on to the usual pump. Check if +5V and GND are in the same place on the 6-wire ribbon cable, as on my schematics. In that case it should be "safe" to try my device.
It would also be fun to flash their own ESP32 with custom firmware. Or replace it with your own. Fun, but too much work to do the same thing :-)

**Reply by wyliecoyoteuk** [reply]
Voltages are the same, but the display doesn't light up.
The web page shows the temperature, but nothing works.

**Reply by visualapproach** [reply]
Did you select the right model in the code?

**Reply by wyliecoyoteuk** [reply]
Correction, dry joint on unit.
Now it works, but the display keeps flashing and changing.
I presume because the built in wifi is trying to update it and conflicting with the add-in.
The unit now works fine on my other unit, 2018 Hawaii model 54154.
If I disconnect the built in module, I get E15

**Reply by visualapproach** [reply]
If you put 560 Ohms resistors between the LLC and display the blinking will likely disappear. (No resistors on 5V and GND ofc)
This is reported several times for newer pumps.

**Reply by GSzabados** [reply]
@wyliecoyoteuk, thanks for those pictures. I was really interested how the WiFi model works. And as @visualapproach stated, that is a really cute solution.

Otherwise of you do use Home Assistant, then there is a custom integration for the Lay-Z-Spa Wifi devices which uses their cloud servers to communicate with the pump.

https://community.home-assistant.io/t/lay-z-spa-hot-tub-wi-fi-pump-automation/229006

**Comment by wyliecoyoteuk** [comment]
I'll be swapping the pumps when I next change the water, can see then.

**Reply by wyliecoyoteuk** [reply]
The wifi in the pump is actually an ESP8266, be interesting to see if it is an additional module.
Correction ESP32.

**Reply by wyliecoyoteuk** [reply]
It is not simple flashing, the characters on the display are being corrupted, then recovering.
The web interface seems to be working.
I have installed the unit in my spare pump, which is now in use, so I will make another one to investigate this, probably in the next week or so, I have spare components available.

**Reply by wyliecoyoteuk** [reply]
I would prefer a simple web interface to the IOT Bestway one that talks to a server in China.

**Comment by strutings** [comment]
I have a brand new Milan 2021 (S100201). 6 wires comes from the pump and 4 wires comes from display. 5V and GND are on the same place as my previous setup (Riviera 2021 S100101) but the display will not boot up. 

I also have built in wifi, but prefer the esp-solution if possible

**Reply by visualapproach** [reply]
Hi @strutings,
I like that you prefer the ESP over OEM, thanks!
Can you post pictures? I don't fully understand how you mean. Where do the wires go to? Isn't the display connected to the pump?

**Reply by strutings** [reply]
![0D81BE13-F0EC-41CC-BC4F-B68A70590E36](https://user-images.githubusercontent.com/3373593/144137252-710db0a2-3e0d-4ad1-9439-5931b32124fc.jpeg)

Here is the connectors between lcd and pump. Four wires runs from display to connector and six down to the pump. 

As far as i can see i have 5V on pin 1, GND on 2 and rx/tx on 3 and 4. Whats makes this extra confusing is the 5V between pin 1 and 6..

**Reply by visualapproach** [reply]
That's weird. Why do they put unused wires when they could've saved a cent there? I don't know why it doesn't work. If you've tried the different models and can't "take control" maybe they changed the setup completely.

**Reply by strutings** [reply]
My theory is they have six wires from lcd to esp32 and four from there to the pump and “re-using” old units. I will have a look at it in a few days. Right now it’s snowing to much to open it..

**Reply by ghost** [reply]
@strutings did you make any progress with this? I've got the exact same problem.
